### PR TITLE
Update branching, cherry-picking, and tagging docs

### PR DESCRIPTION
**Pull request by zjs**
Improve examples and clarify instructions.

---

VIC Appliance Checklist:
- [x] Up to date with `master` branch
- [ ] Added tests
- [x] Considered impact to upgrade
- [x] Tests passing
- [x] Updated documentation
- [ ] Impact assessment checklist

If this is a feature or change to existing functionality, consider areas of impact with the [Impact
Assessment Checklist](https://github.com/vmware/vic-product/blob/master/installer/docs/CHANGE.md)

### CODE REVIEW

File: installer/docs/RELEASE.md [hunk 1]
```diff
@@ -1,100 +1,103 @@
...
-git remote update
-git checkout upstream/master
-git tag -a v1.3.0-dev bbbbbbb
-git pubsh upstream v1.3.0-dev
-```
+### Tagging
 
-After the release candidate has passed QA and the team is ready to release, tag the commit in the
-release branch (`v1.2.0`) and push the tag to Github.
+On the `master` branch, tag the commit for the first release candidate. On the
```
Comment (hickeng) [inline installer/docs/RELEASE.md]: I'm not sure this should be on `master`. If it's a `Z` release then we're already on a release branch. If it's a `Y` release then we've been creating a release branch from master (we've not yet done an `X` release). We're tagging master with `vX.Y.0-dev` but don't recall that we're also tagging the branch point.

File: installer/docs/RELEASE.md [hunk 2]
```diff
@@ -1,100 +1,103 @@
...
+git checkout upstream/releases/X.Y.Z
+git tag -a vX.Y.Z-rc1
+git push upstream vX.Y.Z-rc1
 ```
 
-### Point releases
+```
+git remote update
+git checkout upstream/master
+git tag -a vA.B.C-dev
```
Comment (hickeng) [inline installer/docs/RELEASE.md]: `A.B.C` is a little confusing - I'd suggest either using `X.Y++.0-dev` (or similar) or using specific numeric examples.

File: installer/docs/RELEASE.md [hunk 3]
```diff
@@ -1,100 +1,103 @@
...
+git pubsh upstream vA.B.C-dev
```
Comment (hickeng) [inline installer/docs/RELEASE.md]: `git push`.

File: installer/docs/RELEASE.md (hunk 1 above)
Comment (zjs) [reply installer/docs/RELEASE.md]: Yeah, it looks like this section was not completely updated to match the process from the "Branching" section above.

File: installer/docs/RELEASE.md (hunk 2 above)
Comment (zjs) [reply installer/docs/RELEASE.md]: I opted not to use `X.Y++.0` as I _think_ that's a specific case (albeit the most common one). For example, we might decide to skip to `X++.0.0`.

**Pull request merged**